projects/client/RabbitMQ.Client/src/client/impl/ConsumerWorkService.cs

```diff
@@ -68,11 +62,21 @@ public void AddWork(IModel model, Action fn)
 
         public void RegisterKey(IModel model)
         {
+            if (model.ChannelNumber == 0)
```

Comment:
For those confused by this magic constant: channel 0 is used for "special purpose" protocol communication.